### PR TITLE
Add GitHub Action to load all DAG scripts locally

### PR DESCRIPTION
**Pull request by will-cromar**
# Description

Load DAGs to catch basic Python errors (syntax, missing attributes, etc).

Adds `requirements.txt` to use within GitHub Actions. This has to be manually kept in sync with Terraform unless we can read `requirements.txt` into our templates somehow.

Added some missing dependencies to `requirements.txt` and Terraform.

# Tests

Tested locally. This is expected to fail due to #5:

```
Traceback (most recent call last):
  File "/usr/local/google/home/wcromar/build/ml-auto-solutions/dags/xlml/pytorchxla_torchvision.py", line 45, in <module>
    ).run()
  File "/usr/local/google/home/wcromar/build/ml-auto-solutions/apis/task.py", line 69, in run
    provision, queued_resource, ssh_keys = self.provision()
  File "/usr/local/google/home/wcromar/build/ml-auto-solutions/apis/task.py", line 93, in provision
    if self.task_test_config.custom_tpu_name.strip():
AttributeError: 'JSonnetTpuVmTest' object has no attribute 'custom_tpu_name'
```

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run one-shot tests and provided workload links above if applicable. 
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

**Comment by will-cromar** [comment]
Moving to draft while I work out some GHA plumbing.

**Comment by will-cromar** [comment]
Now the workflow fails where it's supposed to. Ready for review.

File: deployment/cloud_composer_template.tf [hunk 1]
```diff
@@ -102,11 +102,15 @@ resource "google_composer_environment" "example_environment" {
       airflow_config_overrides = {
         core-allowed_deserialization_classes = ".*"
       }
+      # Note: keep this in sync with .github/requirements.txt
       pypi_packages = {
         apache-airflow-providers-sendgrid = ""
-        jsonlines                         = ""
         fabric                            = ""
+        google-cloud-bigquery             = ""
+        google-cloud-storage              = ""
```
Comment (RissyRan) [inline deployment/cloud_composer_template.tf:110]: I think we have google-cloud-bigquery & google-cloud-storage included?

https://cloud.google.com/composer/docs/concepts/versioning/composer-versions

**Review by RissyRan — COMMENTED** [review]
Thanks Will! Just a minor comment.

File: deployment/cloud_composer_template.tf (hunk 1 above)
Comment (will-cromar) [reply deployment/cloud_composer_template.tf:110]: Got it, that's why they weren't listed here. We do need to include them in `.github/requirements.txt`, so do you think they're worth including here too? Without a version specified, I expect this to be a no-op.

Comment (will-cromar) [reply deployment/cloud_composer_template.tf:110]: For some reason, Composer comes with the GPU version of TensorFlow by default, so `tensorflow-cpu` may cause issues. I'll comment these out and leave a note that these packages are pre-installed.

**Review by RissyRan — APPROVED** [review]
Thanks!

**Pull request merged**